ethereum_history_api/circuits/lib/src/merkle_patricia_proofs/proof.nr

```diff
@@ -23,6 +25,41 @@ pub fn leaf_get_len<MAX_LEN>(leaf: Leaf<MAX_LEN>) -> u32 {
     leaf_unpadded_len as u32
 }
 
+// Prefixes mean:
+// 0 - Extension node, even number of nibbles
+// 1 - Extension node, odd number of nibbles
+// 2 - Leaf node, even number of nibbles
+// 3 - Leaf node, odd number of nibbles
+pub fn parity(prefix: u8) -> u8 {
+    assert(prefix < 4, "Prefix must be 0, 1, 2 or 3");
+    let result = prefix % 2;
+    assert((result == EVEN) | (result == ODD), "Invalid parity");
```

Comment:
Is this assert necessary?

Reply:
I think it protects us from developer mistakes like changing values of EVEN and ODD constants or returning sth other than 0/1 as parity. I'm not sure though, what do you think @LogvinovLeon?

Reply:
Seems like an overkill. I'll remove it. Also - const names don't represent the values
And the statement - smth is even or smth is odd is always true